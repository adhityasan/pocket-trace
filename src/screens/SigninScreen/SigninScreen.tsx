import { useState } from "react";
import { StyleSheet, Text, View } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";

import { AuthStackProps } from "@@types/navigations/auth";
import AppIcon from "@components/AppIcon";
import { Button } from "@components/Buttons";
import TextInput from "@components/Inputs/TextInput";
import { PressableText } from "@components/Texts";
import HeadingText from "@components/Texts/HeadingText";
import useAuth from "@hooks/useAuth";
import { SCREEN_NAMES } from "@services/constants/screen";
import globalStyles from "@styles/_global";

const SigninScreen = ({ navigation }: AuthStackProps<SCREEN_NAMES.SIGNIN>) => {
  const [email, setEmail] = useState<string>("");
  const [password, setPassword] = useState<string>("");
  const { isLoading, signin, error } = useAuth();

  const onSubmit = () => {
    signin(email, password);
  };

  return (
    <SafeAreaView style={globalStyles.screenView}>
      <View style={styles.container}>
        <View style={styles.contentHead}>
          <AppIcon />
          <HeadingText type="h1">Pocket Trace</HeadingText>
        </View>
        <View>
          <HeadingText type="h2" style={{ marginBottom: 10 }}>
            Sign In
          </HeadingText>
          <TextInput
            label="Email"
            autoComplete="email"
            value={email}
            onChangeText={setEmail}
            clearable
          />
          <TextInput
            label="Password"
            autoComplete="password"
            secureTextEntry
            value={password}
            defaultValue={password}
            onChangeText={setPassword}
          />
          {error ? <Text style={globalStyles.errorText}>{error.message}</Text> : null}
          <Button
            label="Sign In"
            icon="user-check"
            labelStyle={{ fontWeight: "bold" }}
            disabled={isLoading}
            onPress={onSubmit}
          />
          <View style={styles.otherAuthOptions}>
            <PressableText style={styles.otherAuthOptionText}>Forgot Password ?</PressableText>
            <Text>or</Text>
            <PressableText
              style={styles.otherAuthOptionText}
              onPress={() => navigation.navigate(SCREEN_NAMES.SIGNUP)}>
              Create Account
            </PressableText>
          </View>
        </View>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    paddingHorizontal: 20,
    paddingVertical: 30
  },
  contentHead: {
    alignItems: "center",
    marginBottom: 30
  },
  otherAuthOptions: {
    alignItems: "center"
  },
  otherAuthOptionText: {
    fontWeight: "bold",
    marginVertical: 10
  }
});

export default SigninScreen;
