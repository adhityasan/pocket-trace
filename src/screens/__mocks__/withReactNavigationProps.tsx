import React from "react";

export const createTestScreenProps = (props?: object) => ({
  navigation: {
    state: { params: {} },
    dispatch: jest.fn(),
    goBack: jest.fn(),
    dismiss: jest.fn(),
    navigate: jest.fn(),
    openDrawer: jest.fn(),
    closeDrawer: jest.fn(),
    toggleDrawer: jest.fn(),
    getParam: jest.fn(),
    setParams: jest.fn(),
    addListener: jest.fn(),
    push: jest.fn(),
    replace: jest.fn(),
    pop: jest.fn(),
    popToTop: jest.fn(),
    isFocused: jest.fn()
  },
  ...props
});

const withReactNavigationProps = (Component: any) => {
  const navigationProps = createTestScreenProps();

  return <Component {...navigationProps} />;
};

export default withReactNavigationProps;
