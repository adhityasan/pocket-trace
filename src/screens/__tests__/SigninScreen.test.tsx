import { create, act } from "react-test-renderer";

import SigninScreen from "@screens/SigninScreen";
import withReactNavigationProps from "@screens/__mocks__/withReactNavigationProps";

describe("screens/SigninScreen", () => {
  it("should match snapshot", () => {
    const withNavigationProps = withReactNavigationProps(SigninScreen);
    const result = create(withNavigationProps).toJSON();
    act(() => {
      expect(result).toMatchSnapshot();
    });
  });
});
