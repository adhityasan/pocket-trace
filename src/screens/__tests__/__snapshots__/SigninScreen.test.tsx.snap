// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/SigninScreen should match snapshot 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": "#fafafa",
      "flex": 1,
      "position": "relative",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "paddingHorizontal": 20,
        "paddingVertical": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 30,
        }
      }
    >
      <RNCMaskedView>
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <Image
            source={1}
            style={
              Array [
                Object {
                  "height": 50,
                  "width": 50,
                },
                Object {},
              ]
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#5145a0",
                "height": 50,
                "width": 50,
              },
              undefined,
            ]
          }
        />
      </RNCMaskedView>
      <Text
        style={
          Object {
            "color": "#5145a0",
            "fontSize": 25,
            "fontWeight": "bold",
            "marginBottom": 3,
          }
        }
      >
        Pocket Trace
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#5145a0",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      >
        Sign In
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "justifyContent": "flex-start",
            "marginVertical": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "500",
              "textAlign": "left",
            }
          }
        >
          Email
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#d4cef7",
              "borderColor": "#5145a0",
              "borderRadius": 6,
              "display": "flex",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoComplete="email"
            autoCorrect={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Email"
            placeholderTextColor="rgba(81, 69, 160, 0.7)"
            secureTextEntry={false}
            style={
              Object {
                "flexDirection": "row",
                "fontSize": 16,
                "paddingHorizontal": 10,
                "paddingVertical": 14,
                "width": "90%",
              }
            }
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "display": "flex",
            "justifyContent": "flex-start",
            "marginVertical": 10,
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "500",
              "textAlign": "left",
            }
          }
        >
          Password
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#d4cef7",
              "borderColor": "#5145a0",
              "borderRadius": 6,
              "display": "flex",
              "flexDirection": "row",
              "width": "100%",
            }
          }
        >
          <TextInput
            autoCapitalize="none"
            autoComplete="password"
            autoCorrect={false}
            defaultValue=""
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Password"
            placeholderTextColor="rgba(81, 69, 160, 0.7)"
            secureTextEntry={true}
            style={
              Object {
                "flexDirection": "row",
                "fontSize": 16,
                "paddingHorizontal": 10,
                "paddingVertical": 14,
                "width": "90%",
              }
            }
            value=""
          />
        </View>
      </View>
      <View
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginVertical": 5,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#5145a0",
                "borderRadius": 6,
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "padding": 12,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fafafa",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "textAlign": "center",
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            Sign In
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#5145a0",
                },
                Object {
                  "fontWeight": "bold",
                  "marginVertical": 10,
                },
              ]
            }
          >
            Forgot Password ?
          </Text>
        </View>
        <Text>
          or
        </Text>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#5145a0",
                },
                Object {
                  "fontWeight": "bold",
                  "marginVertical": 10,
                },
              ]
            }
          >
            Create Account
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
